shopify-client/src/views/ProductDetails.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div>
+    <p v-if="error">{{ error }}</p>
+    <div v-if="product" class="text-center">
+      <b-container>
+        <b-row>
+          <b-col sm="6">
+            <b-row>
+              <b-col sm="12">
+                <b-img v-bind:src="product.image"></b-img>
+              </b-col>
+            </b-row>
+          </b-col>
+          <b-col sm="6">
+            <b-row class="text-left">
+              <b-col sm="12">
+                <h1>{{ product.name }}</h1>
+              </b-col>
+              <b-col sm="12">
+                <p>{{ product.description }}</p>
+              </b-col>
+              <b-col sm="12">
+                <p>Price: {{ product.price }}</p>
+              </b-col>
+              <b-col sm="12">
+                <p>Quantity: {{ product.quantity }}</p>
+              </b-col>
+              <b-col sm="12">
+                <b-button v-bind:to="'/app/products/update/' + id"
+                  >Update</b-button
+                >
+              </b-col>
+            </b-row>
+          </b-col>
+        </b-row>
+      </b-container>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "ProductDetails",
+  data: function() {
+    return {
+      product: null,
+      error: "",
+    };
+  },
+  props: ["id"],
+  created: function() {
+    const vm = this;
+    fetch(`/api/products/${vm.id}`)
+      .then((response) => response.json())
+      .then((response) => {
+        if (response.status == 404) {
```

Comment:
The == is done on purpose. The status in the response is a string field. Otherwise I would need to do === "404". In this case, I was taking advantage of the type-ignorant comparison.

Reply:
Surprised its a string since HttpStatus was returned in the controller.